uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestRightsTest.php

```diff
@@ -15,130 +13,103 @@ class CRM_HRLeaveAndAbsences_Service_LeaveRequestRightsTest extends BaseHeadless
   use CRM_HRLeaveAndAbsences_LeaveManagerHelpersTrait;
 
   /**
-   * @var \CRM_HRLeaveAndAbsences_Service_LeaveRequestRights
-   */
-  private $leaveRequestRights;
-
-  /**
-   * @var array
-   *   Contact array
+   * @var int
    */
   private $leaveContact;
 
-  /**
-   * @var array
-   *   Contact array
-   */
-  private $manager;
-
   public function setUp() {
-    $leaveManagerService = new  LeaveManagerService();
-    $this->leaveRequestRights = new LeaveRequestRightsService($leaveManagerService);
-
-    $this->leaveContact = ContactFabricator::fabricate();
-    $this->manager = ContactFabricator::fabricate();
-    $this->setContactAsLeaveApproverOf($this->manager, $this->leaveContact);
+    $this->leaveContact = 1;
+    $this->registerCurrentLoggedInContactInSession($this->leaveContact);
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = [];
   }
 
   public function testCanCancelForReturnsTrueWhenCurrentUserIsLeaveRequestContact() {
-    $this->registerCurrentLoggedInContactInSession($this->leaveContact['id']);
-    CRM_Core_Config::singleton()->userPermissionClass->permissions = [];
-    $this->assertTrue($this->leaveRequestRights->canCancelFor($this->leaveContact['id']));
+    $this->assertTrue($this->getLeaveRightsService()->canCancelFor($this->leaveContact));
   }
 
   public function testCanCancelForReturnsTrueWhenCurrentUserIsLeaveManager() {
-    $this->registerCurrentLoggedInContactInSession($this->manager['id']);
-    CRM_Core_Config::singleton()->userPermissionClass->permissions = [];
-    $this->assertTrue($this->leaveRequestRights->canCancelFor($this->leaveContact['id']));
+    $isAdmin = false;
```

Comment:
Instead of declaring `$isAdmin` and `$isManager` on every test, do this:
- create a `getLeaveRequestRightsForAdminAsCurrentUser()` (which will call `getLeaveRequestRights` passing true to `$isAdmin`)
- create a `getLeaveRequestRightsForLeaveManagerAsCurrentUser()` (which will call `getLeaveRequestRights` passing true to `$isManager`)

Then use these methods on the tests